### PR TITLE
[add] Add support for ACL new

### LINKED ISSUE #1322: Add support for ACL new `AUTH username password`

### PR DESCRIPTION
**Pull request by filipecosta90**
fixes #1322 by adding `AuthACL` as a command, and by properly parsing the url with the user info

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -187,6 +192,7 @@ func ParseURL(redisURL string) (*Options, error) {
 	}
 
 	if u.User != nil {
+		o.Username = u.User.Username()
```
Comment (platinummonkey) [inline options.go:195]: As an aside I’m not sure this functionality ever worked for pre-6 with url.Parse as a plain schema://username:password@host:port/path if only one is given it is assumed to be username not password.

**Comment by filipecosta90** [comment]
> Also is it possible to add a DialOption?

sure =) 
please revise the latest changes

**Comment by platinummonkey** [comment]
bump on this @filipecosta90 :D happy to help contribute

File: options.go (hunk 1 above)
Comment (platinummonkey) [reply options.go:195]: ah, the docs declared `:` prefix, so this would work, resolving

**Comment by filipecosta90** [comment]
> bump on this @filipecosta90 :D happy to help contribute

@platinummonkey applied the go fmt. are we missing anything ?

**Comment by platinummonkey** [comment]
@filipecosta90 only the linter is complaining at the moment:

```
golangci-lint run
options.go:43: line is 121 characters (lll)
	// Use the specified Username to authenticate the current connection with one of the connections defined in the ACL 
```

**Comment by filipecosta90** [comment]
> @filipecosta90 only the linter is complaining at the moment:
> 
> ```
> golangci-lint run
> options.go:43: line is 121 characters (lll)
> 	// Use the specified Username to authenticate the current connection with one of the connections defined in the ACL 
> ```

revising :)

**Comment by platinummonkey** [comment]
> revising :)

submitted a PR to your branch with the changes @filipecosta90

**Comment by filipecosta90** [comment]
> > revising :)
> 
> submitted a PR to your branch with the changes @filipecosta90
> [filipecosta90#1](https://github.com/filipecosta90/redis-1/pull/1)

merged @platinummonkey :+1:

**Comment by platinummonkey** [comment]
@vmihailenco I can verify these changes worked for us on a redis 6.0.1 sentinel cluster with ACLs enabled and one without ACLs enabled. 

logs with ACL, without ACL was the same end result
```$ ./verify -uri sentinel://myuser:superSecret@redis-sentinel:26379/0
Using options: addrs=[redis-sentinel:26379] username=myuser password=superSecret
testing SET
redis: 2020/05/20 18:49:21 sentinel.go:473: sentinel: discovered new sentinel="xxxxxxxx" for master="redis-ha"
redis: 2020/05/20 18:49:21 sentinel.go:473: sentinel: discovered new sentinel="xxxxxxxx" for master="redis-ha"
redis: 2020/05/20 18:49:21 sentinel.go:441: sentinel: new master="redis-ha" addr="xxx.xxx.xxx.xxx:6379"
testing GET
SUCCESS!
```

Any chance we can merge this soon?

**Comment by vmihailenco** [comment]
I am going to revert changes in ring.go since you are supposed to use `RingOptions.NewClient` now - but the rest looks really good. Thanks!

**Comment by ramonberrutti** [comment]
@vmihailenco Can we merge this into v7?

**Comment by vmihailenco** [comment]
@ramonberrutti I believe it is already in v7.

**Comment by ramonberrutti** [comment]
@vmihailenco I'm trying to use Username with Ring Cluster but now I notice that in the master branch is missing too.

**Comment by vmihailenco** [comment]
@ramonberrutti I've added Username as a part of https://github.com/go-redis/redis/pull/1361

**Pull request merged**